exercises/variables/variables4.rs

```diff
@@ -1,10 +1,8 @@
 // variables4.rs
 // Execute `rustlings hint variables4` or use the `hint` watch subcommand for a hint.
 
-// I AM NOT DONE
-
 fn main() {
-    let x = 3;
+    let mut x = 3;
     println!("Number {}", x);
     x = 5; // don't change this line
```

Comment:
I have always embraced the idea of immutability when offered it. Knowing what I know now I would have used `let`.

```suggestion
    let x = 3;
    println!("Number {}", x);
    let x = 5; // don't change this line
```

While this is trivial because its just int32s I would usually follow this approach unless the cost to clone the object was greater than the risk to modify the object.

As soon as you provide a clear immutable constrain I start thinking FP style and when you want to edit you are really clone and replacing it.

I am just gunna keep my mind open and see what I learn next :)